File: 공소연/programmers/P12906.js
// 문제 링크: https://school.programmers.co.kr/learn/courses/30/lessons/12906?language=javascript
// 시간: 42.93ms
// 메모리: 83.4MB

function solution(arr)
{
    const answer = [];
    let lastSeen = null;
    
    arr.forEach(item => {
        if (item !== lastSeen) {
            answer.push(item);
        }
        lastSeen = item;
    })
    
    return answer;
}

File: 공예진/programmers/P12906.java
//문제 링크: https://school.programmers.co.kr/learn/courses/30/lessons/12906
//시간: 0.25ms
//메모리: 73.4MB

import java.util.*;

public class Solution {
    public int[] solution(int[] arr) {
        Stack<Integer> stack = new Stack<>();

        for (int i = 0; i < arr.length; i++) {
            if (stack.isEmpty() || stack.peek() != arr[i]) {
                stack.push(arr[i]);
            }
        }

        int[] answer = new int[stack.size()];
        for (int i = stack.size() - 1; i >= 0; i--) {
            answer[i] = stack.pop();
        }

        return answer;
    }
}